src-a1/main/java/com/example/a1/controller/rest/QuestionController.java

```diff
@@ -0,0 +1,60 @@
+package com.example.a1.controller.rest;
+
+import com.example.a1.dto.QuestionDTO;
+import com.example.a1.dto.TagDTO;
+import com.example.a1.entity.Tag;
+import com.example.a1.service.QuestionManagementService;
+import com.example.a1.service.UserManagementService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RestController
+@RequiredArgsConstructor
+public class QuestionController {
+    private final QuestionManagementService questionManagementService;
+    private final UserManagementService userManagementService;
+
+    @GetMapping("/questions")
+    public List<QuestionDTO> listQuestions() {
+        return questionManagementService.listQuestions();
+    }
+
+    @PostMapping("/questions")
+    public QuestionDTO addQuestion(@RequestBody QuestionDTO question) {
+        Set<Tag> tags = question.getTags().stream().map(Tag::new).collect(Collectors.toSet());
+        return questionManagementService.addQuestion(userManagementService.loadCurrentUser().getId(), question.getTitle(), question.getText(), tags);
+    }
+
+    @GetMapping("/questions/findByTags/{tagsAsString}")
+    public List<QuestionDTO> filterByTag(@PathVariable String tagsAsString) {
```

Comment:
It would be better to use `@RequestParam`s for filter parameters (and use the same `/questions` get mapping). https://www.baeldung.com/spring-request-param